quinn-proto/src/connection/spaces.rs

```diff
@@ -39,6 +39,8 @@ pub(super) struct PacketSpace {
     /// Transmitted but not acked
     // We use a BTreeMap here so we can efficiently query by range on ACK and for loss detection
     pub(super) sent_packets: BTreeMap<u64, SentPacket>,
+    /// Packets that were deemed lost
+    pub(super) lost_packets: BTreeMap<u64, SentPacket>,
```

Comment:
Do we need the full `SentPacket` structure here? It looks like the `time_sent` field is the only one we look at. `SentPacket` isn't always trivial to clone, so storing only that field could save some overhead.